tests/helpers/test_entity.py

```diff
@@ -1657,3 +1658,56 @@ async def async_will_remove_from_hass(self):
     assert len(result) == 2
     assert len(ent.added_calls) == 3
     assert len(ent.remove_calls) == 2
+
+
+def test_entity_description_as_dataclass(snapshot: SnapshotAssertion):
+    """Test EntityDescription behaves like a dataclass."""
+
+    obj = entity.EntityDescription("blah", device_class="test")
+    with pytest.raises(dataclasses.FrozenInstanceError):
+        obj.name = "mutate"
+    with pytest.raises(dataclasses.FrozenInstanceError):
+        delattr(obj, "name")
+
+    assert obj.key == "blah"
+    assert obj.device_class == "test"
+    assert obj == entity.EntityDescription("blah", device_class="test")
+    assert repr(obj) == snapshot
+
+
+def test_extending_entity_description(snapshot: SnapshotAssertion):
+    """Test extending entity descriptions."""
+
+    @dataclasses.dataclass(frozen=True)
+    class FrozenEntityDescription(entity.EntityDescription):
+        extra: str = None
+
+    obj = FrozenEntityDescription("blah", extra="foo", name="name")
+    assert obj.key == "blah"
+    assert obj.extra == "foo"
+    assert obj.name == "name"
+    assert obj == FrozenEntityDescription("blah", extra="foo", name="name")
+    assert repr(obj) == snapshot
```

Comment:
Why only compare the `repr` with the snapshot and not use obj in the snapshot test?

Reply:
I don't understand the comment, can you elaborate a bit?

Reply:
```suggestion
    assert obj == snapshot
    assert repr(obj) == snapshot
```